pkg/controller/pod_volume_backup_controller.go

```diff
@@ -213,6 +219,28 @@ func (c *podVolumeBackupController) processBackup(req *arkv1api.PodVolumeBackup)
 		req.Spec.Tags,
 	)
 
+	// if this is azure, set resticCmd.Cmd().Env to be os.Environ() plus azure.GetResticEnvVars
+	var env []string
+	if strings.HasPrefix(req.Spec.RepoIdentifier, "azure") {
+		// get the BSL named req.Spec.BackupStorageLocation
+		loc, err := c.backupLocationLister.BackupStorageLocations(req.Namespace).Get(req.Spec.BackupStorageLocation)
+		if err != nil {
+			return c.fail(req, errors.Wrap(err, "error getting backup storage location").Error(), log)
+		}
+
+		azureVars, err := azure.GetResticEnvVars(loc.Spec.Config)
+		if err != nil {
+			return c.fail(req, errors.Wrap(err, "error getting azure restic env vars").Error(), log)
+		}
+
+		env = os.Environ()
+		for k, v := range azureVars {
+			env = append(env, fmt.Sprintf("%s=%s", k, v))
+		}
+
+		resticCmd.Env = env
```

Comment:
I notice you set Env here, but also pass env in to GetSnapshotID which, sets Env. I feel like we can just remove this and let the call to GetSnapshotID do the work.

Reply:
I see now, this is needed for both the `GetSnapshotID` AND the later call to Cmd.